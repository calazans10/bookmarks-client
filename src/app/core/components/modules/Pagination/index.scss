@import 'styles/utilities';

.pagination {
  display: flex;
  align-items: center;
  justify-content: center;
  list-style: none;

  li {
    margin-right: rem(9);
  }

  li.previous {
    margin-right: auto;
  }

  li.next {
    margin-right: 0;
    margin-left: auto;
  }

  li:nth-last-child(2) {
    margin-right: 0;
  }

  li a {
    color: #2980b9;
    font-family: 'Gilroy SemiBold';
    font-size: em(14);
    line-height: lh(14px, 17px);
    cursor: pointer;

    &:focus,
    &:hover,
    &:active {
      color: #20638f;
    }
  }

  li.selected a {
    display: flex;
    align-items: center;
    justify-content: center;
    width: rem(25);
    height: rem(25);
    border: rem(1) solid #2980b9;
    border-radius: rem(10);
    font-weight: bold;
  }

  li.previous a,
  li.next a {
    display: flex;
    align-items: center;
    font-size: em(12);
    text-transform: uppercase;
  }

  li.previous.disabled a,
  li.next.disabled a {
    color: #afbbd9;
    cursor: default;
    pointer-events: none;
  }
}
