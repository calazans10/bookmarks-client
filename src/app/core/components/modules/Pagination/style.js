import ReactPaginate from 'react-paginate';
import styled from 'styled-components';
import { em, rem } from 'polished';

export const Paginate = styled(ReactPaginate)`
  display: flex;
  align-items: center;
  justify-content: center;
  list-style: none;

  li {
    margin-right: ${rem('9px')};
  }

  li.previous {
    margin-right: auto;
  }

  li.next {
    margin-right: 0;
    margin-left: auto;
  }

  li:nth-last-child(2) {
    margin-right: 0;
  }

  li a {
    color: #2980b9;
    font-family: 'Gilroy SemiBold';
    font-size: ${em('14px')};
    line-height: lh(14px, 17px);
    cursor: pointer;

    &:focus,
    &:hover,
    &:active {
      color: #20638f;
    }
  }

  li.selected a {
    display: flex;
    align-items: center;
    justify-content: center;
    width: ${rem('25px')};
    height: ${rem('25px')};
    border: ${rem('1px')} solid #2980b9;
    border-radius: ${rem('10px')};
    font-weight: bold;
  }

  li.previous a,
  li.next a {
    display: flex;
    align-items: center;
    font-size: ${em('12px')};
    text-transform: uppercase;
  }

  li.previous.disabled a,
  li.next.disabled a {
    color: #afbbd9;
    cursor: default;
    pointer-events: none;
  }
`;
