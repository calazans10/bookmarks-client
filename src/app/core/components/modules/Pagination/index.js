import React from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import { Container } from './style';

const Pagination = ({ initialPage, pageCount, onChange }) => (
  <Container>
    <ReactPaginate
      previousLabel="< Previous"
      nextLabel="Next >"
      initialPage={initialPage}
      pageCount={Math.ceil(pageCount)}
      containerClassName="pagination"
      onPageChange={onChange}
    />
  </Container>
);

Pagination.propTypes = {
  initialPage: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Pagination;
