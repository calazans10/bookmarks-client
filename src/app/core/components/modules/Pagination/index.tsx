import React from 'react';
import ReactPaginate from 'react-paginate';
import { Container } from './style';

type PaginationProps = {
  initialPage: number;
  count: number;
  limit: number;
  total: number;
  onChange: (selectedItem: { selected: number }) => void;
};

const Pagination = ({ initialPage, count, limit, total, onChange }: PaginationProps) => {
  const isVisible = count <= total && total > limit;

  if (!isVisible) {
    return null;
  }

  return (
    <Container>
      <ReactPaginate
        previousLabel="< Previous"
        nextLabel="Next >"
        initialPage={initialPage}
        pageCount={Math.ceil(total / limit)}
        marginPagesDisplayed={2}
        pageRangeDisplayed={5}
        containerClassName="pagination"
        onPageChange={onChange}
      />
    </Container>
  );
};

export default Pagination;
