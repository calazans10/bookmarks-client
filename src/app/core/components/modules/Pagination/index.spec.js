import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination from './index';

describe('Pagination', () => {
  it('renders without crashing', () => {
    const props = {
      initialPage: 0,
      pageCount: 10,
      onChange: jest.fn(),
    };
    const wrapper = shallow(<Pagination {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
