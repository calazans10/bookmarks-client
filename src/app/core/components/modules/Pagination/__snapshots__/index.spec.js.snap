// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders without crashing 1`] = `
<PaginationBoxView
  activeClassName="selected"
  breakLabel="..."
  containerClassName="pagination"
  disableInitialCallback={false}
  disabledClassName="disabled"
  initialPage={0}
  marginPagesDisplayed={3}
  nextClassName="next"
  nextLabel="Next >"
  onPageChange={[MockFunction]}
  pageCount={10}
  pageRangeDisplayed={2}
  previousClassName="previous"
  previousLabel="< Previous"
/>
`;
